src/Umbraco.Web.BackOffice/Filters/CheckIfUserTicketDataIsStaleAttribute.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Options;
+using Umbraco.Core;
+using Umbraco.Core.BackOffice;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Configuration.Models;
+using Umbraco.Core.Mapping;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Membership;
+using Umbraco.Core.Services;
+using Umbraco.Extensions;
+using Umbraco.Web.BackOffice.Security;
+using Umbraco.Web.Common.Security;
+
+namespace Umbraco.Web.BackOffice.Filters
+{
+    internal sealed class CheckIfUserTicketDataIsStaleAttribute : TypeFilterAttribute
+    {
+        public CheckIfUserTicketDataIsStaleAttribute() : base(typeof(CheckIfUserTicketDataIsStaleFilter))
+        {
+        }
+
+        private class CheckIfUserTicketDataIsStaleFilter : IAsyncActionFilter
+        {
+            private readonly IRequestCache _requestCache;
+            private readonly UmbracoMapper _umbracoMapper;
+            private readonly IUserService _userService;
+            private readonly IEntityService _entityService;
+            private readonly ILocalizedTextService _localizedTextService;
+            private readonly IOptions<GlobalSettings> _globalSettings;
+            private readonly BackOfficeSignInManager _backOfficeSignInManager;
+            private readonly IBackOfficeAntiforgery _backOfficeAntiforgery;
+
+            public CheckIfUserTicketDataIsStaleFilter(
+                IRequestCache requestCache,
+                UmbracoMapper umbracoMapper,
+                IUserService userService,
+                IEntityService entityService,
+                ILocalizedTextService localizedTextService,
+                IOptions<GlobalSettings> globalSettings,
+                BackOfficeSignInManager backOfficeSignInManager,
+                IBackOfficeAntiforgery backOfficeAntiforgery)
+            {
+                _requestCache = requestCache;
+                _umbracoMapper = umbracoMapper;
+                _userService = userService;
+                _entityService = entityService;
+                _localizedTextService = localizedTextService;
+                _globalSettings = globalSettings;
+                _backOfficeSignInManager = backOfficeSignInManager;
+                _backOfficeAntiforgery = backOfficeAntiforgery;
+            }
+
+
+            public async Task OnActionExecutionAsync(ActionExecutingContext actionContext, ActionExecutionDelegate next)
+            {
+                await CheckStaleData(actionContext);
+
+                await next();
+
+                await CheckStaleData(actionContext);
+
+                //we need new tokens and append the custom header if changes have been made
```

Comment:
Might be better as a method name with a comment on the method (future change)

Reply:
I agree, just migrated as close to one to one as possible